pygmt/figure.py

```diff
@@ -241,6 +242,12 @@ def psconvert(self, icc_gray=False, **kwargs):
             else:
                 kwargs["N"] += "+i"
 
+        # check if the parent directory exists
+        if kwargs.get("F") and not Path(kwargs.get("F")).parent.exists():
+            raise FileNotFoundError(
+                f"No such directory: '{Path(kwargs.get('F')).parent}'"
```

Comment:
```suggestion
            directory = Path(kwargs.get("F")).parent
            raise FileNotFoundError(
                f"No such directory: '{directory}', please "
                f"create it first, e.g. using `os.makedirs(name='{directory}')`"
```

Reply:
Here is the error message from matplotlib:
```python
FileNotFoundError: [Errno 2] No such file or directory: 'anypathh/image.png'
```
I think we can follow mpl and don't need to say how to create the directory.

Reply:
Prefer having an error message that not only lets the user know what is the issue (missing a directory) but also how to resolve it (create the directory). We could shorten the message though like so:
```suggestion
            directory = Path(kwargs.get("F")).parent
            raise FileNotFoundError(
                f"No such directory: '{directory}', please create it first."
```
and assume that the user knows how to create the folder using `os.makedirs` or just manually in their file manager.